docs/tutorials/python/sagas/index.md

```diff
@@ -0,0 +1,551 @@
+from shared import TASK_QUEUE_NAME
+```
+
+
+The Worker is a crucial component that executes the defined Workflows and Activities.
+Setting up the Worker correctly will ensure that your system can process tasks efficiently and reliably.
+
+In the `main()` function, you will specify how to connect to the Temporal server, create a Worker, and run it.
+This Worker will listen to the specified Task Queue and execute the defined Workflows and Activities.
+
+```python
+interrupt_event = asyncio.Event()
+
+
+async def main():
+    # Connect to the Temporal server
+    client = await Client.connect("localhost:7233")
+
+    # Create a Worker that listens to the specified task queue
+    worker = Worker(
+        client,
+        task_queue=TASK_QUEUE_NAME,
+        workflows=[BookWorkflow],
+        activities=[
+            book_car,
+            book_hotel,
+            book_flight,
+            undo_book_car,
+            undo_book_hotel,
+            undo_book_flight,
+        ],
+    )
+    await worker.run()
+    try:
+        await interrupt_event.wait()
+    finally:
+        print("\nShutting down the worker\n")
+
+
+if __name__ == "__main__":
+    loop = asyncio.get_event_loop()
+    try:
+        loop.run_until_complete(main())
+    except KeyboardInterrupt:
+        print("\nInterrupt received, shutting down...\n")
+        interrupt_event.set()
+        loop.run_until_complete(loop.shutdown_asyncgens())
+```
+
+The `Client.connect()` line connects to the Temporal server running on `localhost` at port `7233`.
+This can be modified to run a Worker on Temporal Cloud.
+
+The `Worker` is initialized with the client, the task queue name, the list of Workflows, and the list of Activities.
+This setup ensures the Worker knows which tasks to listen for and execute.
+The `await worker.run()` line starts the Worker, making it ready to receive tasks and execute the corresponding Activities and Workflows.
+
+**Start the Worker**
+
+To start the Worker, run the following command in your terminal:
+
+```bash
+python run_worker.py
+```
+
+Once the Worker is running, it will be ready to execute Workflows and Activities as tasks are submitted to the specified task queue.
+
+Now that the Worker is set up and running, you can set up the Client to initiate the booking process.
+
+## Create the Client
+
+The Client is responsible for initiating the Workflow and handling the booking process.
+Setting up the Client will allow you to interact with the Temporal server and trigger the booking Workflow.
+
+Create a new file named `run_workflow.py`:
+
+First, import the necessary modules, including `uuid`, Flask, and Temporal `Client`.
+
+```python
+import uuid
+
+from flask import Flask, jsonify, request
+from temporalio.client import Client
+
+from activities import BookVacationInput
+from saga_workflows import BookWorkflow
+from shared import TASK_QUEUE_NAME
+```
+
+Initialize the Flask app and set up the dependency injection for the Temporal client.
```

Comment:
This could use better explanation. Why do I care about dependency injection, and why do I need to do it? 

What thing are we *really* doing here? We're making the Temporal Client available to all of our Flask routes in a way that doesn't create a new instance of the Temporal client on each web request. This lets you share resources.

The dependency injection is just an implementation detail. Keep asking "why am I doing this" and then add the answer to the tutorial. :)